src/autolabel/data_loaders/read_datasets.py

```diff
@@ -0,0 +1,215 @@
+from typing import Dict, List, Union
+
+import logging
+import pandas as pd
+from dataclasses import dataclass
+from datasets import Dataset
+from sqlalchemy.sql.selectable import Selectable
+from autolabel.configs import AutolabelConfig
+
+logger = logging.getLogger(__name__)
+from typing import Union
+
+
+@dataclass
+class DataAttribute:
+    """Data Attributes"""
+
+    columns: List
+    dataset: Union[pd.DataFrame, Dataset]
```

Comment:
this can be just a dataframe type (i don't think we ever pass in the raw HF dataset)